### ISSUE TITLE
HTTP response body was broken, if struct to be converted to JSON has boolean value.

### ISSUE THREAD

**Issue by shu-mutou** [emicklei/go-restful#449]
v2.13.0 goes well, but 2.14.0 or later version occurs this issue.

**Comment by emicklei** [comment]
thank you for reporting this. I will investigate what is going on.

**Comment by emicklei** [comment]
did you try v2.14.2  as it has fixed w.r.t 2.14.0

**Comment by shu-mutou** [comment]
I'd tried 2.14.0 and 2.14.2

**Comment by emicklei** [comment]
thanks, do you have a test that shows this problem?

**Comment by shu-mutou** [comment]
There are no results such as unit tests, but I suspect it is due to the use of the `rune` function.

**Comment by shu-mutou** [comment]
Additional info: I found that changing `EnableContentEncoding` from `true` to `false` with 2.14.2 works well.

**Comment by shu-mutou** [comment]
I tried v3, v3.2.0 works but v3.3.1 does not.

**Comment by emicklei** [comment]
@shu-mutou does it fail for the same reason?

**Comment by shu-mutou** [comment]
@emicklei Yes. Completely same.

**Comment by emicklei** [comment]
@shu-mutou  can you have a look at the example i just added (v3 branch).

With content encoding enabled + accept header , you can gzipped content of the users list.

**Comment by shu-mutou** [comment]
@emicklei thank you for the example!

In our implementation, I found the responses were gzipped twice.
And I commented out following, then it seems to work fine for our implementation.
https://github.com/emicklei/go-restful/blob/master/container.go#L240-L257

I guess check like following is needed for `dispatch` in `container.go` like in `ServeHTTP` or `Handle`.
https://github.com/emicklei/go-restful/blob/master/container.go#L305-L309
